### PR TITLE
[IMP] l10n_es: New taxes, taxes codes and fiscal positions according law changes

### PR DESCRIPTION
**Pull request by pedrobaeza**
- v5.2: Añadida retención 19,5% arrendamientos.
- v5.1: Renombrado todo lo relacionado con arrendamientos para no incluir la
  palabra "IRPF", ya que no es como tal IRPF.
- v5.0: Se ha rehecho toda la parte de impuestos para dar mayor facilidad de
  consulta de los datos para las declaraciones de la AEAT y para cubrir todas
  las casuísticas fiscales españolas actuales. Éstas son las características
  más destacadas:
  - Desdoblamiento de los impuestos principales para bienes y para servicios.
  - Nuevo árbol de códigos de impuestos orientado a cada modelo de la AEAT.
  - Nuevos códigos para los códigos de impuestos para facilitar su
    actualización.
  - La casilla del modelo viene ahora en la descripción, no en el código.
  - Posiciones fiscales ajustadas para el desdoblamiento.
  - Nuevo impuesto y posición fiscal para retención IRPF 19%.
  - Nuevo impuesto para revendedores con recargo de equivalencia.
  - Nuevas posiciones fiscales para retenciones de arrendamientos.
  - Pequeños ajustes en cuentas contables.

### CODE REVIEW

**Comment by pedrobaeza** [comment]
@odony Please check this that we talked about on Odoo experience.

@rafaelbn, @antespi, @alejandrosantana, @cubells, @anajuaristi, please review it to give your opinion. It's a verbatim copy of l10n-spain one removing name_get overwrite that Odoo won't accept in core.

**Comment by odony** [comment]
@pedrobaeza Thanks for the PR!

@qdp-odoo Could you or someone from your team have a look at this PR and see if we could port this to 9.0 without too much trouble? I expect conflicts galore when forward-porting. Perhaps a separate PR for 9.0 would be ideal, and we'd drop everything during the forward-port?

**Comment by pedrobaeza** [comment]
Tell me if I can help in the porting.

**Comment by pedrobaeza** [comment]
Any advance on this? I'll have some little final adjustments (some changes on "Allow reconciliation" flag or internal type), so tell me where to propose this little changes (or may I do it here)

**Comment by rafaelbn** [comment]
Reviewed, tested. Already using in 12 different customers.
:+1: 
@odony @qdp-odoo  any news in this point? Imposible to use v9 in Spain without this.

**Comment by rafaelbn** [comment]
Please, @qdp-odoo could you review this ? Check https://github.com/odoo/odoo/pull/8453#issuecomment-138918413 from @odony Thanks a lot

**Comment by qdp-odoo** [comment]
Hello,

i don't know how I didn't see that PR... But someone here will take care of reviewing this for 8.0 and making a port for 9.0.

Thanks for the contribution,
Quentin

**Comment by pedrobaeza** [comment]
Thank you very much, @qdp-odoo, please let me know if you need help in something.

**Comment by pedrobaeza** [comment]
Any news about this?

**Comment by rafaelbn** [comment]
Hi @odony @qsm-odoo , please could you merge this PR in v8? we have problems in OCA, runbot untill this is merged. Thanks

**Comment by odony** [comment]
@rafaelbn: yes, @qdp-odoo is working on it as he said, no need to notify other people like @qsm-odoo who has nothing to do with this ;-)

**Comment by JavierIniesta** [comment]
Reviewed and tested :+1:

**Comment by JMODaERP** [comment]
Tested and O.K. :+1:

**Comment by pvy-odoo** [comment]
Hi, I have reviewed your PR.
I have a few notes before it can be merged :
Why go to version 8.0.5.2.0 ? You should just go to version 5.0
I don't think we need the migration scripts. Can you explain why it's absolutely needed ?

Thanks for your contribution.

Pierre

**Comment by pedrobaeza** [comment]
Hi, Pierre, thanks for your review. 

The version is that one because we have been evolving the module in the spanish localization (https://github.com/OCA/l10n-spain/tree/8.0/l10n_es), and we follow the OCA's version numbering convention: https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#version-numbers. There has been some minor adjustments, and that's why we use the second version number. We even have another minor revision since I propose this PR, so I can include it here if you want.

The 8.0 prefix is because Odoo doesn't prefix the main version number if it's already present, so we use in OCA the full version notation to distinguish between Odoo versions when we only have the module code/ZIP without context. As Odoo is package all together, I think it can be omitted, but anyway it doesn't hurt.

MIgration scripts helps a lot to those that already have the chart of accounts installed, because it prepare the data to be automatically detected by _account_chart_update_ module (https://github.com/OCA/account-financial-tools/tree/8.0/account_chart_update) to update all the data. In case of a pure Odoo installation, the migration path is not fully completed without this module, so I'm not sure what is your strategy with current Spanish customers that wants to upgrade their chart of accounts, so y

**Comment by pedrobaeza** [comment]
Sorry, bad key combination...

File: addons/l10n_es/data/account_type.xml [hunk 1]
```diff
@@ -4,13 +4,11 @@
     <!-- Actualizar método de cierre de tipos de cuenta estándar -->
     
     <data noupdate="1">
-        <record id="account.data_account_type_income"
-            model="account.account.type">
+        <record id="account.data_account_type_income" model="account.account.type">
             <field name="name">Income</field>
```
Comment (qdp-odoo) [inline addons/l10n_es/data/account_type.xml]: your patch does not modify the existing behavior of these lines but I really wonder why you need them: don't you reset your P&L at the beginning of a new fiscal year? 

If yes, I don't get how you're dealing with the closing/opening entries given the close method of income/expense accounts set to balance.

**Comment by qdp-odoo** [comment]
all fine for me

**Comment by rafaelbn** [comment]
Hi, I think this one is ready to review and mege as @qdp-odoo says is :+1: 
Could you inform us? Thanks a lot!

**Comment by pedrobaeza** [comment]
I have added latest changes (a new tax)

**Comment by pedrobaeza** [comment]
@odony, can you give the latest push to this change in v8?

**Comment by rafaelbn** [comment]
Hi @pvy-odoo  , @qdp-odoo said OK
Why this PR is blocked? What do you need?
Thanks in advance!
cc @odony

**Comment by pedrobaeza** [comment]
cc @Tecnativa

**Comment by Yenthe666** [comment]
@odony and @mart-e could you guys give some feedback here please?

**Comment by rafaelbn** [comment]
:pray:

**Comment by mart-e** [comment]
Hello,

Sorry for the delay and thanks for the changes and reviews. I have merged the commits at 

e3e5cfe [IMP] l10n_es: Added a new tax "IVA soportado no sujeto"
730890f [IMP] l10n_es: New taxes, taxes codes and fiscal positions according law changes

I have changed the version numbers because, it may cause problems during forward port. When reaching the version 9.0, you version number will no longer match and you would bump.
I prefer to keep things clean and keep the simple version numbers.

**Comment by pedrobaeza** [comment]
Thank you very much! In the coming days, we will make another change to include other new taxes (complementary to the new ones). Can I ping you to get these new ones also merged?

**Comment by mart-e** [comment]
Sure, as always for localisation changes, we would love reviews from external contributors too

**Pull request closed without merging**